jOOQ/src/main/java/org/jooq/tools/jdbc/JDBCUtils.java

```diff
@@ -106,7 +114,9 @@ public static final SQLDialect dialect(Connection connection) {
 
 
                 String url = m.getURL();
-                result = dialect(url);
+                int majorVersion = m.getDatabaseMajorVersion();
+                int minorVersion = m.getDatabaseMinorVersion();
```

Comment:
From experience with (and distrust of) JDBC drivers, I'll wrap these in a try-catch-ignore block :)

Reply:
Okay :) I'll take on mind this!